plugin/completion.py

```diff
@@ -1,35 +1,85 @@
 import sublime
 import sublime_plugin
-from .core.completion import parse_completion_response, format_completion
-from .core.configurations import is_supported_syntax
-from .core.documents import get_document_position, position_is_word
-from .core.edit import parse_text_edit
+
+from .core.protocol import Request, Range, InsertTextFormat
+from .core.settings import settings, client_configs
 from .core.logging import debug
-from .core.protocol import Request
+from .core.completion import parse_completion_response, format_completion
 from .core.registry import session_for_view, client_from_session, LSPViewEventListener
+from .core.configurations import is_supported_syntax
+from .core.documents import get_document_position
 from .core.sessions import Session
-from .core.settings import settings, client_configs
+from .core.edit import parse_text_edit
+from .core.views import range_to_region
 from .core.typing import Any, List, Dict, Tuple, Optional, Union
 
 
-class CompletionState(object):
-    IDLE = 0
-    REQUESTING = 1
-    APPLYING = 2
-    CANCELLING = 3
+class LspSelectCompletionItemCommand(sublime_plugin.TextCommand):
```

Comment:
One way to test these changes thoroughly is to have a unit test that runs this command.

e.g. something like this

```python
import sublime
from unittesting import DeferrableTestCase

class SelectCompletionItemTest(DeferrableTestCase):

    def setUp(self) -> Generator:
        super().setUp()
        self.window = sublime.active_window()
        self.view = self.window.new_file()
        yield lambda: not self.view.is_loading()
        self.view.set_scratch(True)

    def tearDown(self) -> None:
        self.view.close()
        super().tearDown()

    def test_a(self) -> None:
        self.view.run_command("insert", {"characters": "a"})
        item = { ... }
        self.view.run_command("lsp_select_completion", {"item": item})
        actual = self.view.substr(sublime.Region(0, self.view.size()))
        self.assertEqual("b", actual)